src/osversion/VersionTranslator.cpp

```diff
@@ -8258,6 +8336,105 @@ namespace osversion {
 
         ss << object;
 
+      } else if (iddname == "OS:Boiler:HotWater") {
```

Comment:
I'm confused why BoilerHotWater is here. Is this branch just slightly out of date?

Reply:
I just moved it up in the `if-else` ladder.

Handle objects first, then do the lookup in the map things.